services/core/Darksky/darksky/agent.py

```diff
@@ -0,0 +1,475 @@
+        else:
+            raise ValueError('Invalid location. Expected format is: '
+                             '"{"lat": "xxx.xxxx", "long": "xxx.xxxx"}"')
+        grequest = [grequests.get(url, verify=requests.certs.where(),
+                                  headers=self.headers, timeout=3)]
+        gresponse = grequests.map(grequest)[0]
+        if gresponse is None:
+            raise RuntimeError("get request did not return any "
+                               "response")
+        try:
+            response = json.loads(gresponse.content)
+            return response
+        except ValueError:
+            self.generate_response_error(url, gresponse.status_code)
+
+    def format_multientry_response(self, location, response, service):
+        """
+        Used to extract the data not used by the RPC method, and store it in
+        the cache, helping to limit the number of API calls used to obtain data
+        :param location: location dictionary to include with cached data
+        :param response: Darksky forecast response
+        :param service:
+        :return: formatted response data by service
+        """
+        data = []
+        generation_time = get_aware_utc_now()
+        if service == "get_minutely_forecast":
+            interval_minutes = self.get_update_interval(
+                'get_minutely_forecast').seconds / 60
+            gen_minutes = (generation_time.minute / interval_minutes)\
+                * interval_minutes
+            generation_time = format_timestamp(generation_time.replace(
+                microsecond=0, second=0, minute=gen_minutes))
+        else:
+            generation_time = format_timestamp(generation_time.replace(
+                microsecond=0, second=0, minute=0))
+        for entry in response['data']:
+            entry_time = datetime.datetime.fromtimestamp(entry['time'],
+                                                         pytz.utc)
+            if 'forecast' in service:
+                data.append([json.dumps(location), generation_time, entry_time,
+                             json.dumps(entry)])
+            else:
+                data.append([json.dumps(location), entry_time, json.dumps(
+                    entry)])
+        return data
+
+    def query_current_weather(self, location):
+        """
+        Retrieve data from the Darksky API, return formatted current data and
+        store forecast data in cache
+        :param location: location dictionary requested by the user
+        :return: Timestamp and data for current data from the Darksky API
+        """
+        darksky_response = self.get_darksky_forecast(
+            SERVICES_MAPPING['SERVICE_CURRENT_WEATHER']['service'], location)
+        current_response = darksky_response.pop('currently')
+        # Darksky required attribution
+        current_response["attribution"] = "Powered by Dark Sky"
+        current_time = datetime.datetime.fromtimestamp(
+            current_response['time'], pytz.utc)
```

Comment:
Did you check if the time conversion happens right? I remember that that python’s datetime has issues converting naive timestamp to UTC correctly. That is why we use a different package (from tzlocal import get_localzone. Refer to test_historian or actuator agent) Is the time returned naive. Does the response specify the timezone info. If api response has timezone then you can create a date time object with the specific tzinfo and then convert to UTC.